client/src/main/java/io/pravega/client/tables/impl/TableSegment.java

```diff
@@ -0,0 +1,177 @@
+ * previously existed or not, and regardless of what that Key's version is.
+ * * Conditional Updates will only overwrite an existing value if the specified version matches that Key's version. If
+ * the key does not exist, the {@link TableKey} or {@link TableEntry} must have been created with {@link KeyVersion#NOT_EXISTS}
+ * in order for the update to succeed.
+ * * Unconditional Removals will remove a Key regardless of what that Key's version is. The operation will also succeed (albeit
+ * with no effect) if the Key does not exist.
+ * * Conditional Removals will remove a Key only if the specified {@link TableKey#getVersion()} matches that Key's version.
+ * It will also fail (with no effect) if the Key does not exist and Version is not set to {@link KeyVersion#NOT_EXISTS}.
+ *
+ * @param <KeyT>   Table Key Type.
+ * @param <ValueT> Table Value Type.
+ */
+@Beta
+public interface TableSegment<KeyT, ValueT> extends AutoCloseable {
+
+    /**
+     * Inserts or updates an existing Table Entry into this Table Segment.
+     * @param entry The Entry to insert or update. If {@link TableEntry#getKey()#getVersion()} is null, this will perform
+     *              an Unconditional Update, otherwise it will perform a Conditional Update based on the information
+     *              provided. See {@link TableSegment} doc for more details on Types of Updates.
+     * @return A CompletableFuture that, when completed, will contain the {@link KeyVersion} associated with the newly
+     * inserted or updated entry. Notable exceptions:
+     * <ul>
+     * <li>{@link ConditionalTableUpdateException} If this is a Conditional Update and the condition was not satisfied.
+     * </ul>
+     */
+    CompletableFuture<KeyVersion> put(TableEntry<KeyT, ValueT> entry);
+
+    /**
+     * Inserts new or updates existing Table Entries into this Table Segment.
+     *
+     * @param entries A Collection of entries to insert or update. If for at least one such entry,
+     *                {@link TableEntry#getKey()#getVersion()} returns a non-null value, this will perform an atomic
+     *                Conditional Update where all the entries either get applied or none will; otherwise a Unconditional
+     *                Update will be performed. See {@link TableSegment} doc for more details on Types of Updates.
+     * @return A CompletableFuture that, when completed, will contain a map of {@link KeyT} to {@link KeyVersion} which
+     * represents the versions for the inserted/updated keys. Notable exceptions:
+     * <ul>
+     * <li>{@link ConditionalTableUpdateException} If this is a Conditional Update and the condition was not satisfied.
+     * This exception will contain the Keys that failed the validation.
+     * </ul>
+     */
+    CompletableFuture<Map<KeyT, KeyVersion>> put(Collection<TableEntry<KeyT, ValueT>> entries);
+
+    /**
+     * Removes the given key from this Table Segment.
```

Comment:
This is documented in this interface's Javadoc. If a conditional remove, it will fail, otherwise it will go through (unconditional updates/removes will succeed regardless of the prior state of that key).